testing_scripts/run.py

```diff
@@ -0,0 +1,195 @@
+import argparse
+import os
+import shutil
+from typing import Tuple
+
+# Helper Functions that we created and run
+from prepare import prepare
+from add_lfind import invoke_lfind
+from run_lfind import run_lfind
+from process_results import process_results, clean_up_project, write_to_csv
+
+# These are the defaul values, can be modified by user/tester to represent how they want lfind ran
+def set_lfind_parameters(args):
+    quickchick = "Unset Lfind Quickchick.\n" if args.no_quickchick else ""
+    proverbot = "Unset Lfind Proverbot.\n" if args.no_proverbot else ""
+    synth_size = f"Set Lfind BatchSize \"{args.synth_batch_size}\".\n" if args.synth_batch_size != 6 else ""
+    # Here we just check if it's myth or not because we only consider one other synthesizer
+    synthesizer = f"Set Lfind Synthesizer \"{args.synthesizer}\".\n" if args.synthesizer == "myth" else ""
+    timeout = f"Set Lfind Batch-Size \"{args.synth_timeout}\".\n" if args.synth_timeout != 12 else ""
+    return f"{quickchick}{proverbot}{synthesizer}{synth_size}{timeout}"
+
+def parse_arguments() -> Tuple[argparse.Namespace, argparse.ArgumentParser]:
+    parser = argparse.ArgumentParser(
+        description=
+        "Run Benchmark Files")
+    parser.add_argument('--parent', default=None)
+    parser.add_argument('--result_folder', default=None)
+    parser.add_argument('--file', default=None)
+    parser.add_argument('--project', default=None)
+    parser.add_argument('--benchmarks', default=None)
+    parser.add_argument('--clean', default=False, action='store_true')
+    parser.add_argument('--no_quickchick', default=False, action='store_true')
+    parser.add_argument('--no_proverbot', default=False, action='store_true')
+    parser.add_argument('--synth_batch_size', default=6, type=int)
+    parser.add_argument('--synth_timeout', default=12, type=int)
+    parser.add_argument('--synthesizer', default="coqsynth")
+    parser.add_argument('--rerun', default=False, action='store_true')
+    parser.add_argument('--lfind_files', default=None)
+    return parser.parse_args()
+
+def run_on_project(project_path,log_directory,cwd,parameters):
+    # Run all pre-processing
+    helper_lemmas, all_lemmas = prepare(project_path,cwd)
+    if helper_lemmas == None:
+        return None, None, None, None
+    # Create folders for each of the examples
+    lfind_folders, lfind_folder = invoke_lfind(helper_lemmas,project_path,lfind_parameters=parameters)
+    # Set log directory if not provided
+    log_folder = log_directory if log_directory is not None else os.path.join(os.path.dirname(lfind_folder),"RESULTS")
+    if os.path.isdir(log_folder) == False:
+        os.mkdir(log_folder)
+    # Run lfind for each file and get the results
```

Comment:
Similar to the comment before `run_lfind`, we want to run lfind at the granularity of a helper lemma location?

Reply:
I'm not sure what the question is here? The code above that is highlighted is just creating a log directory if one is not already there? Not sure if there is another part you are asking questions about?